On kubemark master, kubelet now runs as a supervisord process and all master components as pods

shyamjvs opened the pull request:
This PR fixes issue #37485
On kubemark, previously we had a custom setup that runs master components as top-level processes under supervisord, which is lighter than running kubelet and docker.
This PR makes kubelet run as a process under supervisord and the master components (apiserver, controller-manager, scheduler, etcd) as pods, making testing on kubemark mimic real clusters better.
Also, start-kubemark-master.sh now closely resembles cluster/gce/gci/configure-helper.sh, allowing easy integration in future.

cc @kubernetes/sig-scalability @wojtek-t @gmarek

shyamjvs committed:
On kubemark master, kubelet now runs as a supervisord process and all master components as pods

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/38227)
<!-- Reviewable:end -->

gmarek commented on test/kubemark/resources/manifests/etcd-events.yaml:
```diff
@@ -0,0 +1,58 @@
+      name: varetcd
+    - mountPath: /var/log/etcd-events.log
+      name: varlogetcd
+  dnsPolicy: ClusterFirst
```
please move top level arguments to the top of the file.

gmarek commented on test/kubemark/resources/manifests/etcd-events.yaml:
```diff
@@ -0,0 +1,58 @@
+    name: varetcd
+  - hostPath:
+      path: /var/log/etcd-events.log
+    name: varlogetcd
```
Some newline encoding problem - please fix.

gmarek commented on test/kubemark/resources/manifests/etcd-events.yaml:
```diff
@@ -0,0 +1,58 @@
+      protocol: TCP
+    resources:
+      limits:
+        cpu: 100m
```
this is most certainly not enough. Please remove limit.

gmarek commented on test/kubemark/resources/manifests/etcd-events.yaml:
```diff
@@ -0,0 +1,58 @@
+      requests:
+        cpu: 100m
+    volumeMounts:
+    - mountPath: /var/etcd
```
/var/etcd/events

gmarek commented on test/kubemark/resources/manifests/etcd.yaml:
```diff
@@ -0,0 +1,55 @@
+  terminationGracePeriodSeconds: 30
+  volumes:
+  - hostPath:
+      path: /mnt/master-pd/var/etcd
```
I'm not sure what exactly should be there, but rather not this - you'll have both etcd's using the same directory, which is bad. My guess would be `/var/etcd/events`, assuming that it's mounted first.

gmarek replied:
/var/etcd

gmarek commented on test/kubemark/resources/manifests/etcd.yaml:
```diff
@@ -0,0 +1,55 @@
+      path: /mnt/master-pd/var/etcd
+    name: varetcd
+  - hostPath:
+      path: /var/log/etcd.log
```
delete etcd.log

gmarek commented on test/kubemark/resources/manifests/kube-controller-manager.yaml:
```diff
@@ -0,0 +1,58 @@
+      timeoutSeconds: 15
+    name: kube-controller-manager
+    resources:
+      limits:
```
remove limits

gmarek commented on test/kubemark/start-kubemark-master.sh:
```diff
@@ -16,9 +16,219 @@
+# on the master -- etcd's data, a few settings, and security certs/keys/tokens.
+# safe-format-and-mount only formats an unformatted disk, and mkdir -p will
+# leave a directory be if it already exists.
+function mount-master-pd {
```
we had an option to put event etcd on a separate disk - was it removed, or it's still somewhere?

gmarek commented on test/kubemark/start-kubemark-master.sh:
```diff
@@ -16,9 +16,219 @@
+
+	# Decide whether to enable a docker registry mirror. This is taken from
+	# the "kube-env" metadata value.
+	if [[ -n "${DOCKER_REGISTRY_MIRROR_URL:-}" ]]; then
```
I don't think we'll be using it - please remove.

gmarek commented on test/kubemark/start-kubemark-master.sh:
```diff
@@ -16,9 +16,219 @@
+function assemble-docker-flags {
+	echo "Assemble docker command line flags"
+	local docker_opts="-p /var/run/docker.pid --iptables=false --ip-masq=false"
+	if [[ "${TEST_CLUSTER:-}" == "true" ]]; then
```
It's always a test cluster:)

gmarek commented on test/kubemark/start-kubemark-master.sh:
```diff
@@ -16,9 +16,219 @@
+		docker_opts+=" --registry-mirror=${DOCKER_REGISTRY_MIRROR_URL}"
+	fi
+
+	echo "DOCKER_OPTS=\"${docker_opts} ${EXTRA_DOCKER_OPTS:-}\"" > /etc/default/docker
```
We never set "EXTRA_DOCER_OPTS", so there's no need of reading it here

gmarek commented on test/kubemark/start-kubemark-master.sh:
```diff
@@ -27,129 +237,204 @@ stdout_logfile=/var/log/${name}.log
-supervisorctl reread
-supervisorctl update
+# Set required path variables and run config script.
+KUBE_HOME="${KUBE_ROOT}/kubernetes"
```
move directory definitions to the beginning of the file

gmarek commented on test/kubemark/start-kubemark-master.sh:
```diff
@@ -16,9 +16,219 @@
+		flags+=" --pod-cidr=${MASTER_IP_RANGE}"
+	fi
+	# Network plugin
+	if [[ -n "${NETWORK_PROVIDER:-}" ]]; then
```
I don't think we care about this or any of options below.

gmarek commented on test/kubemark/start-kubemark-master.sh:
```diff
@@ -27,129 +237,204 @@ stdout_logfile=/var/log/${name}.log
+  cp "${KUBE_ROOT}/etcd.yaml" /etc/kubernetes/manifests
+
+  prepare-log-file /var/log/etcd-events.log
+  sed -i -e "s@{{instance_prefix}}@${INSTANCE_PREFIX}@g" "${KUBE_ROOT}/etcd-events.yaml"
```
I was using `##` instead of `{{` - I'm fine with anything as long as it's consistent across kubemark codebase.

gmarek commented on test/kubemark/start-kubemark-master.sh:
```diff
@@ -27,129 +237,204 @@ stdout_logfile=/var/log/${name}.log
+		CLOUD_CONFIG_MOUNT="{\"name\": \"cloudconfigmount\",\"mountPath\": \"/etc/gce.conf\", \"readOnly\": true},"
+	fi
+	DOCKER_REGISTRY="gcr.io/google_containers"
+	if [[ -n "${KUBE_DOCKER_REGISTRY:-}" ]]; then
```
we don't need an option to change it for now.

gmarek commented on test/kubemark/start-kubemark-master.sh:
```diff
@@ -16,9 +16,219 @@
+function load-docker-images {
+	echo "Start loading kube-system docker images"
+	local -r img_dir="${KUBE_BINDIR}"
+	try-load-docker-image "${img_dir}/kube-apiserver.tar"
```
where from do you have those tars?

shyamjvs commented:
Review status: 0 of 7 files reviewed at latest revision, 17 unresolved discussions.

---

*[test/kubemark/start-kubemark-master.sh, line 64 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYLUx2L8PwMGHgzOxSW-r1-64:-KYOd4WHVx_8nGnrDRvG:blhxtum) ([raw file](https://github.com/kubernetes/kubernetes/blob/805d824ca392c1cae6e89c0211b3ce315153856c/test/kubemark/start-kubemark-master.sh#L64)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

we had an option to put event etcd on a separate disk - was it removed, or it's still somewhere?
</blockquote></details>

Added code for it.

---

*[test/kubemark/start-kubemark-master.sh, line 106 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYLUx2L8PwMGHgzOxSW-r1-106:-KYNkNf3wIa8Ieowv_5o:bqpmqqx) ([raw file](https://github.com/kubernetes/kubernetes/blob/805d824ca392c1cae6e89c0211b3ce315153856c/test/kubemark/start-kubemark-master.sh#L106)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

It's always a test cluster:)
</blockquote></details>

Done. I kept it thinking of eventually merging this with the real configure-helper.sh script. Too far in future, maybe? :)

---

*[test/kubemark/start-kubemark-master.sh, line 115 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYLUx2L8PwMGHgzOxSW-r1-115:-KYNkoocpyu0UjZZT7u7:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/805d824ca392c1cae6e89c0211b3ce315153856c/test/kubemark/start-kubemark-master.sh#L115)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

I don't think we'll be using it - please remove.
</blockquote></details>

Done.

---

*[test/kubemark/start-kubemark-master.sh, line 120 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYLUx2L8PwMGHgzOxSW-r1-120:-KYNlMdDMxrfrwTzBKiA:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/805d824ca392c1cae6e89c0211b3ce315153856c/test/kubemark/start-kubemark-master.sh#L120)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

We never set "EXTRA_DOCKER_OPTS", so there's no need of reading it here
</blockquote></details>

Done.

---

*[test/kubemark/start-kubemark-master.sh, line 155 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYLUx2L8PwMGHgzOxSW-r1-155:-KYOdKJ5A1576wZGyIfC:b53l9jv) ([raw file](https://github.com/kubernetes/kubernetes/blob/805d824ca392c1cae6e89c0211b3ce315153856c/test/kubemark/start-kubemark-master.sh#L155)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

where from do you have those tars?
</blockquote></details>

Those tars are right beside their corresponding binaries inside the server tar (which is created each time on 'make quick-release') in the location kubernetes/server/bin. So it seems like they are the right ones.

---

*[test/kubemark/start-kubemark-master.sh, line 203 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYLUx2L8PwMGHgzOxSW-r1-203:-KYNoNibzg2FF6wYC8dF:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/805d824ca392c1cae6e89c0211b3ce315153856c/test/kubemark/start-kubemark-master.sh#L203)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

I don't think we care about this or any of options below.
</blockquote></details>

Done.

---

*[test/kubemark/start-kubemark-master.sh, line 276 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYLUx2L8PwMGHgzOxSW-r1-276:-KYNoQrSEJcViGi1yGdr:b-jkjgn3) ([raw file](https://github.com/kubernetes/kubernetes/blob/805d824ca392c1cae6e89c0211b3ce315153856c/test/kubemark/start-kubemark-master.sh#L276)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

I was using `##` instead of `{{` - I'm fine with anything as long as it's consistent across kubemark codebase.
</blockquote></details>

<code>{{</code> is used in the raw manifest files in _output/release-stage/manifests/kubernetes/gci-trusty/. Tried to stick with that convention.

---

*[test/kubemark/start-kubemark-master.sh, line 290 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYLUx2L8PwMGHgzOxSW-r1-290:-KYOSl03t4mA7dnydU9P:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/805d824ca392c1cae6e89c0211b3ce315153856c/test/kubemark/start-kubemark-master.sh#L290)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

We won't have this file.
</blockquote></details>

Done.

---

*[test/kubemark/start-kubemark-master.sh, line 296 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYLUx2L8PwMGHgzOxSW-r1-296:-KYNrOVH1qXWLzUXj3qk:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/805d824ca392c1cae6e89c0211b3ce315153856c/test/kubemark/start-kubemark-master.sh#L296)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

we don't need an option to change it for now.
</blockquote></details>

Done.

---

*[test/kubemark/start-kubemark-master.sh, line 414 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYLUx2L8PwMGHgzOxSW-r1-414:-KYNs0TxrlsXflvyqyRF:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/805d824ca392c1cae6e89c0211b3ce315153856c/test/kubemark/start-kubemark-master.sh#L414)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

move directory definitions to the beginning of the file
</blockquote></details>

Done.

---

*[test/kubemark/start-kubemark-master.sh, line 417 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYLUx2L8PwMGHgzOxSW-r1-417:-KYNs5EY-QGu_ZQAXVNG:b-ve62de) ([raw file](https://github.com/kubernetes/kubernetes/blob/805d824ca392c1cae6e89c0211b3ce315153856c/test/kubemark/start-kubemark-master.sh#L417)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

why you need this?
</blockquote></details>

Because the variable definitions in config-default.sh are not available in kubemark-master, but are needed at places in this file. For eg. ${KUBELET_TEST_ARGS}, ${APISERVER_TEST_ARGS}, ${SCHEDULER_TEST_ARGS}, etc. Is there a better way around this without hard-coding those variables (and hence duplicating their definitions despite having the config-default.sh) in this script?

---

*[test/kubemark/resources/manifests/etcd-events.yaml, line 39 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYLUx2L8PwMGHgzOxSR-r1-39:-KYNuoKo62ELVEvQp9tI:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/805d824ca392c1cae6e89c0211b3ce315153856c/test/kubemark/resources/manifests/etcd-events.yaml#L39)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

this is most certainly not enough. Please remove limit.
</blockquote></details>

Done.

---

*[test/kubemark/resources/manifests/etcd-events.yaml, line 47 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYLUx2L8PwMGHgzOxSR-r1-47:-KYNv6Ps1122RIxRMU7a:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/805d824ca392c1cae6e89c0211b3ce315153856c/test/kubemark/resources/manifests/etcd-events.yaml#L47)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

please move top level arguments to the top of the file.
</blockquote></details>

Done.

---

*[test/kubemark/resources/manifests/etcd-events.yaml, line 54 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYLUx2L8PwMGHgzOxSR-r1-54:-KYO9ukIw-H61a_O2DtN:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/805d824ca392c1cae6e89c0211b3ce315153856c/test/kubemark/resources/manifests/etcd-events.yaml#L54)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

you'll have collision between event and non-event etcd - it should be /var/etcd/events
</blockquote></details>

Done.

---

*[test/kubemark/resources/manifests/etcd-events.yaml, line 58 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYLUx2L8PwMGHgzOxSR-r1-58:-KYO9vKZINHQUx4__nnE:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/805d824ca392c1cae6e89c0211b3ce315153856c/test/kubemark/resources/manifests/etcd-events.yaml#L58)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

Some newline encoding problem - please fix.
</blockquote></details>

Done.

---

*[test/kubemark/resources/manifests/etcd.yaml, line 51 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYLUx2L8PwMGHgzOxSS-r1-51:-KYO9ygtlITRoJzeWaLr:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/805d824ca392c1cae6e89c0211b3ce315153856c/test/kubemark/resources/manifests/etcd.yaml#L51)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

/var/etcd
</blockquote></details>

Done.

---

*[test/kubemark/resources/manifests/kube-controller-manager.yaml, line 26 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYLUx2L8PwMGHgzOxSU-r1-26:-KYO1s0ZRTMCIZTcpzZI:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/805d824ca392c1cae6e89c0211b3ce315153856c/test/kubemark/resources/manifests/kube-controller-manager.yaml#L26)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

remove limits
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227)*
<!-- Sent from Reviewable.io -->

timothysc commented:
/cc @jeremyeder

gmarek commented:
Reviewed 1 of 7 files at r1, 5 of 6 files at r2.
Review status: 6 of 7 files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[test/kubemark/start-kubemark-master.sh, line 106 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYLUx2L8PwMGHgzOxSW-r1-106:-KYT5zRZKB2eVy4qXGJ6:bxnqg89) ([raw file](https://github.com/kubernetes/kubernetes/blob/805d824ca392c1cae6e89c0211b3ce315153856c/test/kubemark/start-kubemark-master.sh#L106)):*
<details><summary><i>Previously, shyamjvs (Shyam JVS) wrote…</i></summary><blockquote>

Done. I kept it thinking of eventually merging this with the real configure-helper.sh script. Too far in future, maybe? :)
</blockquote></details>

As always - first things first. I'm not sure we'll get to configure-helper.sh anytime soon.

---

*[test/kubemark/start-kubemark-master.sh, line 155 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYLUx2L8PwMGHgzOxSW-r1-155:-KYTLQzdWKV-G9Oj81-d:b-o63eqq) ([raw file](https://github.com/kubernetes/kubernetes/blob/805d824ca392c1cae6e89c0211b3ce315153856c/test/kubemark/start-kubemark-master.sh#L155)):*
<details><summary><i>Previously, shyamjvs (Shyam JVS) wrote…</i></summary><blockquote>

Those tars are right beside their corresponding binaries inside the server tar (which is created each time on 'make quick-release') in the location kubernetes/server/bin. So it seems like they are the right ones.
</blockquote></details>

Right - SG.

---

*[test/kubemark/start-kubemark-master.sh, line 276 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYLUx2L8PwMGHgzOxSW-r1-276:-KYT8r7BEwSbbfqlvNc7:bb91ws0) ([raw file](https://github.com/kubernetes/kubernetes/blob/805d824ca392c1cae6e89c0211b3ce315153856c/test/kubemark/start-kubemark-master.sh#L276)):*
<details><summary><i>Previously, shyamjvs (Shyam JVS) wrote…</i></summary><blockquote>

<code>{{</code> is used in the raw manifest files in _output/release-stage/manifests/kubernetes/gci-trusty/. Tried to stick with that convention.
</blockquote></details>

It's because they're taken from the jinja template. It's not that I don't like {{, or that I like ## - I just want to have consistency in kubemark codebase (and it's older that gci) - please change all ## to {{ then.

---

*[test/kubemark/start-kubemark-master.sh, line 38 at r3](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYT4_D29-8v24cHyC86:-KYT4_D29-8v24cHyC87:b-g79ena) ([raw file](https://github.com/kubernetes/kubernetes/blob/795495348884f9936891a3f098b28f0816dd3388/test/kubemark/start-kubemark-master.sh#L38)):*
> ```Shell
> 
> # Compute etcd/event-etcd related variables.
> function compute-etcd-variables {
> ```

This mixes default etcd config with event etcd one - please split them. You should have one common function and two dedicated ones (or if the changes are simple, just do them inline)

---

*[test/kubemark/start-kubemark-master.sh, line 70 at r3](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYT5i9szdOhhYVd94BS:-KYT5i9szdOhhYVd94BT:bhpxw1q) ([raw file](https://github.com/kubernetes/kubernetes/blob/795495348884f9936891a3f098b28f0816dd3388/test/kubemark/start-kubemark-master.sh#L70)):*
> ```Shell
> 
> # Finds the master PD device; returns it in MASTER_PD_DEVICE
> function find-master-pd() {
> ```

It's not master-pd anymore, right? :) Instead of setting a variable you should just `echo` the path.

---

*[test/kubemark/start-kubemark-master.sh, line 170 at r3](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYT8XBOxuOPEarOFPe5:-KYT8XBOxuOPEarOFPe6:bwhauqh) ([raw file](https://github.com/kubernetes/kubernetes/blob/795495348884f9936891a3f098b28f0816dd3388/test/kubemark/start-kubemark-master.sh#L170)):*
> ```Shell
> 	flags+=" --babysit-daemons=true"
> 	flags+=" --cgroup-root=/"
> 	flags+=" --cloud-provider=gce"
> ```

Is this needed?

---

*[test/kubemark/start-kubemark-master.sh, line 204 at r3](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYT8jZsqMXLv2imAxyY:-KYT8jZsqMXLv2imAxyZ:bwykwh) ([raw file](https://github.com/kubernetes/kubernetes/blob/795495348884f9936891a3f098b28f0816dd3388/test/kubemark/start-kubemark-master.sh#L204)):*
> ```Shell
> #   DOCKER_REGISTRY
> function compute-master-manifest-variables {
> 	CLOUD_CONFIG_OPT=""
> ```

IIUC this is always empty.

---

*[test/kubemark/start-kubemark-master.sh, line 268 at r3](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYTL_3dlUlVjrdP_mKa:-KYTL_3dlUlVjrdP_mKb:b-t28m5u) ([raw file](https://github.com/kubernetes/kubernetes/blob/795495348884f9936891a3f098b28f0816dd3388/test/kubemark/start-kubemark-master.sh#L268)):*
> ```Shell
> 
> 	# Calculate variables and assemble the command line.
> 	local params="${APISERVER_TEST_ARGS:-} ${CLOUD_CONFIG_OPT}"
> ```

It would be more readable if it was a separate function (compute-api-server-params or sth) - similarly for other components.

---

*[test/kubemark/start-kubemark-master.sh, line 411 at r3](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYTD7uoUTthfCskZZv_:-KYTD7uoUTthfCskZZva:bm5vhmu) ([raw file](https://github.com/kubernetes/kubernetes/blob/795495348884f9936891a3f098b28f0816dd3388/test/kubemark/start-kubemark-master.sh#L411)):*
> ```Shell
> start-kube-scheduler
> 
> # Wait till all components are working fine.
> ```

this only waits for api-server

---

*[test/kubemark/start-kubemark.sh, line 28 at r3](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYTCE0RSVyRhxqBVyRK:-KYTCE0RSVyRhxqBVyRL:b-ei3pof) ([raw file](https://github.com/kubernetes/kubernetes/blob/fc361206e7d26868ee8f5495009b5a9a4ac5c9df/test/kubemark/start-kubemark.sh#L28)):*
> ```Shell
>   cat > "${RESOURCE_DIRECTORY}/apiserver_flags" <<EOF
> ${APISERVER_TEST_ARGS}
> --storage-backend=${STORAGE_BACKEND}
> ```

All those variables have to be passed down - we set them in our test environments differently for different tests.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227)*
<!-- Sent from Reviewable.io -->

shyamjvs commented:
Review status: 3 of 9 files reviewed at latest revision, 8 unresolved discussions.

---

*[test/kubemark/start-kubemark.sh, line 28 at r3](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYTCE0RSVyRhxqBVyRK:-KYU-V-e1FCHX60Laex4:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/fc361206e7d26868ee8f5495009b5a9a4ac5c9df/test/kubemark/start-kubemark.sh#L28)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

All those variables have to be passed down - we set them in our test environments differently for different tests.
</blockquote></details>

Done.

---

*[test/kubemark/start-kubemark-master.sh, line 276 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYLUx2L8PwMGHgzOxSW-r1-276:-KYT_40_GNCYFtp6tuaM:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/805d824ca392c1cae6e89c0211b3ce315153856c/test/kubemark/start-kubemark-master.sh#L276)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

It's because they're taken from the jinja template. It's not that I don't like {{, or that I like ## - I just want to have consistency in kubemark codebase (and it's older that gci) - please change all ## to {{ then.
</blockquote></details>

Done.

---

*[test/kubemark/start-kubemark-master.sh, line 38 at r3](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYT4_D29-8v24cHyC86:-KYTRiBEPJza7nuUamM8:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/795495348884f9936891a3f098b28f0816dd3388/test/kubemark/start-kubemark-master.sh#L38)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

This mixes default etcd config with event etcd one - please split them. You should have one common function and two dedicated ones (or if the changes are simple, just do them inline)
</blockquote></details>

Done.

---

*[test/kubemark/start-kubemark-master.sh, line 70 at r3](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYT5i9szdOhhYVd94BS:-KYTSyI33avDeNQ57ESf:b-dkvhtt) ([raw file](https://github.com/kubernetes/kubernetes/blob/795495348884f9936891a3f098b28f0816dd3388/test/kubemark/start-kubemark-master.sh#L70)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

It's not master-pd anymore, right? :) Instead of setting a variable you should just `echo` the path.
</blockquote></details>

Done. Good catch!
However, I kept the function name same because it finds a PD device on the master. Is that ok?

---

*[test/kubemark/start-kubemark-master.sh, line 170 at r3](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYT8XBOxuOPEarOFPe5:-KYTaL2ospFHV46UKGQF:b3593rj) ([raw file](https://github.com/kubernetes/kubernetes/blob/795495348884f9936891a3f098b28f0816dd3388/test/kubemark/start-kubemark-master.sh#L170)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

Is this needed?
</blockquote></details>

Removed. Works without them.

---

*[test/kubemark/start-kubemark-master.sh, line 204 at r3](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYT8jZsqMXLv2imAxyY:-KYTYF8LlB6hk_zd0tBO:b-13otsn) ([raw file](https://github.com/kubernetes/kubernetes/blob/795495348884f9936891a3f098b28f0816dd3388/test/kubemark/start-kubemark-master.sh#L204)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

IIUC this is always empty.
</blockquote></details>

Removed it. (Kept it so it would be possible to pass cloud related config, but it's not required for now.)

---

*[test/kubemark/start-kubemark-master.sh, line 268 at r3](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYTL_3dlUlVjrdP_mKa:-KYTiWy2WSlj-osj-7jV:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/795495348884f9936891a3f098b28f0816dd3388/test/kubemark/start-kubemark-master.sh#L268)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

It would be more readable if it was a separate function (compute-api-server-params or sth) - similarly for other components.
</blockquote></details>

Done.

---

*[test/kubemark/start-kubemark-master.sh, line 411 at r3](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYTD7uoUTthfCskZZv_:-KYT_Ooem6o9cMtlcdGw:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/795495348884f9936891a3f098b28f0816dd3388/test/kubemark/start-kubemark-master.sh#L411)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

this only waits for api-server
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227)*
<!-- Sent from Reviewable.io -->

shyamjvs commented:
Review status: 3 of 9 files reviewed at latest revision, 8 unresolved discussions.

---

*[test/kubemark/start-kubemark-master.sh, line 170 at r3](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYT8XBOxuOPEarOFPe5:-KYUN5z27wTwxRHu-Crc:bfg7rep) ([raw file](https://github.com/kubernetes/kubernetes/blob/795495348884f9936891a3f098b28f0816dd3388/test/kubemark/start-kubemark-master.sh#L170)):*
<details><summary><i>Previously, shyamjvs (Shyam JVS) wrote…</i></summary><blockquote>

Removed. Works without them.
</blockquote></details>

Added the flags back as they are needed. Sorry, I didn't figure it out during starting kubemark as healthz passed fine. Apparently, turning off babysit-daemons leads to the following error in kubelet logs:

E1208 16:17:55.903451    2967 pod_workers.go:184] Error syncing pod d85b8f9080365a427819ca0263
9920cf, skipping: failed to "StartContainer" for "kube-controller-manager" with RunContainerEr
ror: "runContainer: Error response from daemon: Cannot start container 656542e1a3d40214538d9d7
4571beb40e2ae1c5362ee28632939698d908c413d: [8] System error: read parent: connection reset by 
peer"

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227)*
<!-- Sent from Reviewable.io -->

gmarek commented:
Reviewed 3 of 7 files at r4.
Review status: 6 of 9 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[test/kubemark/resources/start-kubemark-master.sh, line 68 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYYZmRsTQ9SEYdCjbRB:-KYYZmRsTQ9SEYdCjbRC:b-pke79u) ([raw file](https://github.com/kubernetes/kubernetes/blob/47f41a8a27a42b2dde46f7ad539da750ec8396d2/test/kubemark/resources/start-kubemark-master.sh#L68)):*
> ```Shell
> 
> # Finds a master PD device with name '$1'.
> function find-master-pd() {
> ```

find-attached-pd

---

*[test/kubemark/resources/start-kubemark-master.sh, line 81 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYYZub5NThAX3R103_t:-KYYZub5NThAX3R103_u:by57ynh) ([raw file](https://github.com/kubernetes/kubernetes/blob/47f41a8a27a42b2dde46f7ad539da750ec8396d2/test/kubemark/resources/start-kubemark-master.sh#L81)):*
> ```Shell
> # on the master. safe-format-and-mount only formats an unformatted disk, and
> # mkdir -p will leave a directory be if it already exists.
> function mount-master-pd() {
> ```

mount-pd

---

*[test/kubemark/resources/start-kubemark-master.sh, line 90 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYYZy9Wz5KAdOow6Qyn:-KYYZy9Wz5KAdOow6Qyo:b-d67lkh) ([raw file](https://github.com/kubernetes/kubernetes/blob/47f41a8a27a42b2dde46f7ad539da750ec8396d2/test/kubemark/resources/start-kubemark-master.sh#L90)):*
> ```Shell
> 	fi
> 
> 	echo "Mounting master-pd"
> ```

Mounting PD ${pd_name} at ${mount_point}

---

*[test/kubemark/resources/start-kubemark-master.sh, line 162 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYY_sINuyoZp_eoSBfY:-KYY_sINuyoZp_eoSBfZ:btuwi08) ([raw file](https://github.com/kubernetes/kubernetes/blob/47f41a8a27a42b2dde46f7ad539da750ec8396d2/test/kubemark/resources/start-kubemark-master.sh#L162)):*
> ```Shell
> 	local -r name="kubelet"
> 	local exec_command="${KUBE_BINDIR}/kubelet "
> 	local flags="${KUBELET_TEST_ARGS:-}"
> ```

Move to `compute_kubelet_args` (or sth) function.

---

*[test/kubemark/resources/start-kubemark-master.sh, line 181 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYYa6b2G_rEZRuTxMes:-KYYa6b3um2EsgdCZqmv:b7r6k18) ([raw file](https://github.com/kubernetes/kubernetes/blob/47f41a8a27a42b2dde46f7ad539da750ec8396d2/test/kubemark/resources/start-kubemark-master.sh#L181)):*
> ```Shell
> 	exec_command+="${flags}"
> 
> 	cat >>"/etc/supervisor/conf.d/${name}.conf" <<EOF
> ```

Move to the function.

---

*[test/kubemark/resources/start-kubemark-master.sh, line 296 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYYbSfZN2-kw4KYaFoG:-KYYbSfZN2-kw4KYaFoH:b-i51sxc) ([raw file](https://github.com/kubernetes/kubernetes/blob/47f41a8a27a42b2dde46f7ad539da750ec8396d2/test/kubemark/resources/start-kubemark-master.sh#L296)):*
> ```Shell
> # Assumed vars:
> #   DOCKER_REGISTRY
> function start-kube-apiserver {
> ```

All those functions look basically the same - are they possible to generalize?

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227)*
<!-- Sent from Reviewable.io -->

shyamjvs commented:
Review status: 5 of 9 files reviewed at latest revision, 7 unresolved discussions.

---

*[test/kubemark/resources/start-kubemark-master.sh, line 68 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYYZmRsTQ9SEYdCjbRB:-KYYeWe69QuxRHnZZpBt:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/47f41a8a27a42b2dde46f7ad539da750ec8396d2/test/kubemark/resources/start-kubemark-master.sh#L68)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

find-attached-pd
</blockquote></details>

Done.

---

*[test/kubemark/resources/start-kubemark-master.sh, line 81 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYYZub5NThAX3R103_t:-KYYekdEe37WFc7YzQbM:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/47f41a8a27a42b2dde46f7ad539da750ec8396d2/test/kubemark/resources/start-kubemark-master.sh#L81)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

mount-pd
</blockquote></details>

Done.

---

*[test/kubemark/resources/start-kubemark-master.sh, line 90 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYYZy9Wz5KAdOow6Qyn:-KYYf1j0bH2Q6cQuRHOa:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/47f41a8a27a42b2dde46f7ad539da750ec8396d2/test/kubemark/resources/start-kubemark-master.sh#L90)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

Mounting PD ${pd_name} at ${mount_point}
</blockquote></details>

Done.

---

*[test/kubemark/resources/start-kubemark-master.sh, line 162 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYY_sINuyoZp_eoSBfY:-KYYg93IAdjyi6GqW4x_:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/47f41a8a27a42b2dde46f7ad539da750ec8396d2/test/kubemark/resources/start-kubemark-master.sh#L162)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

Move to `compute_kubelet_args` (or sth) function.
</blockquote></details>

Done.

---

*[test/kubemark/resources/start-kubemark-master.sh, line 181 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYYa6b2G_rEZRuTxMes:-KYYhhvwABv24EHo6gmr:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/47f41a8a27a42b2dde46f7ad539da750ec8396d2/test/kubemark/resources/start-kubemark-master.sh#L181)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

Move to the function.
</blockquote></details>

Done.

---

*[test/kubemark/resources/start-kubemark-master.sh, line 296 at r4](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYYbSfZN2-kw4KYaFoG:-KYYrG5PsYyYjOMzZ56L:b84nxuv) ([raw file](https://github.com/kubernetes/kubernetes/blob/47f41a8a27a42b2dde46f7ad539da750ec8396d2/test/kubemark/resources/start-kubemark-master.sh#L296)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

All those functions look basically the same - are they possible to generalize?
</blockquote></details>

Great observation! Reduced the code size by a few tens of lines. Didn't try doing this before as I planned to stick to the function structure in configure-helper.sh

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227)*
<!-- Sent from Reviewable.io -->

gmarek commented:
One nit and it looks good to me (but still please test it thoroughly)

---

Reviewed 2 of 7 files at r4, 2 of 2 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[test/kubemark/start-kubemark.sh, line 52 at r5](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYZ0I62V7jzYdUxNPzE:-KYZ0I62V7jzYdUxNPzF:b-tape70) ([raw file](https://github.com/kubernetes/kubernetes/blob/38efafe1c77deb44ca825f4582ed0fd85044a3f2/test/kubemark/start-kubemark.sh#L52)):*
> ```Shell
> fi
> 
> cat >> "${RESOURCE_DIRECTORY}/kubemark-master-env.sh" <<EOF
> ```

You can move this above `if`s

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-:-KYZ0xiYUugzcsGM-_gi:b-2h00ic)*
<!-- Sent from Reviewable.io -->

shyamjvs commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[test/kubemark/start-kubemark.sh, line 52 at r5](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-KYZ0I62V7jzYdUxNPzE:-KYZ2cGTvU1tfuMnUizr:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/38efafe1c77deb44ca825f4582ed0fd85044a3f2/test/kubemark/start-kubemark.sh#L52)):*
<details><summary><i>Previously, gmarek (Marek Grabowski) wrote…</i></summary><blockquote>

You can move this above `if`s
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227)*
<!-- Sent from Reviewable.io -->

gmarek commented:
Reviewed 1 of 1 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227)*
<!-- Sent from Reviewable.io -->

wojtek-t commented:
I would also like to take a look before merging. Will take a look on Monday.

wojtek-t commented on test/kubemark/resources/manifests/etcd-events.yaml:
```diff
@@ -0,0 +1,53 @@
+  name: etcd-server-events
+  namespace: kube-system
+spec:
+  dnsPolicy: ClusterFirst
```
please remove

shyamjvs replied:
Done.

wojtek-t commented on test/kubemark/resources/manifests/etcd-events.yaml:
```diff
@@ -0,0 +1,53 @@
+  dnsPolicy: ClusterFirst
+  hostNetwork: true
+  nodeName: {{instance_prefix}}-master
+  restartPolicy: Always
```
Isn't this a default?

shyamjvs replied:
Yes it is. Removed it.

wojtek-t commented on test/kubemark/resources/manifests/etcd-events.yaml:
```diff
@@ -0,0 +1,53 @@
+  hostNetwork: true
+  nodeName: {{instance_prefix}}-master
+  restartPolicy: Always
+  terminationGracePeriodSeconds: 30
```
Is this a default?

shyamjvs replied:
Yes it is. Removed it.

wojtek-t commented on test/kubemark/resources/manifests/etcd.yaml:
```diff
@@ -0,0 +1,52 @@
+  name: etcd-server
+  namespace: kube-system
+spec:
+  dnsPolicy: ClusterFirst
```
Same comments as in the above file.

shyamjvs replied:
Done.

wojtek-t commented on test/kubemark/resources/manifests/kube-apiserver.yaml:
```diff
@@ -0,0 +1,70 @@
+  name: kube-apiserver
+  namespace: kube-system
+spec:
+  dnsPolicy: ClusterFirst
```
Same comments as in the above file.

shyamjvs replied:
Done.

wojtek-t commented on test/kubemark/resources/manifests/kube-controller-manager.yaml:
```diff
@@ -0,0 +1,56 @@
+  name: kube-controller-manager
+  namespace: kube-system
+spec:
+  dnsPolicy: ClusterFirst
```
Same comments as in the above file.

shyamjvs replied:
Done.

wojtek-t commented on test/kubemark/resources/manifests/kube-scheduler.yaml:
```diff
@@ -0,0 +1,39 @@
+  name: kube-scheduler
+  namespace: kube-system
+spec:
+  dnsPolicy: ClusterFirst
```
Same comments as in the above file.

wojtek-t commented on test/kubemark/start-kubemark.sh:
```diff
@@ -22,36 +22,41 @@ KUBE_ROOT=$(readlink -e ${TMP_ROOT} 2> /dev/null || perl -MCwd -e 'print Cwd::ab
+SERVICE_CLUSTER_IP_RANGE=${SERVICE_CLUSTER_IP_RANGE}
+
+# Kubelet related variables.
+MASTER_IP_RANGE=${MASTER_IP_RANGE}
```
This is used only in standalone-kubelet mode. We are not using it in kubemark at all.

shyamjvs replied:
Done.

wojtek-t commented on test/kubemark/start-kubemark.sh:
```diff
@@ -22,36 +22,41 @@ KUBE_ROOT=$(readlink -e ${TMP_ROOT} 2> /dev/null || perl -MCwd -e 'print Cwd::ab
+MASTER_IP_RANGE=${MASTER_IP_RANGE}
+
+# Etcd related variables.
+ETCD_IMAGE=2.2.1
```
=${ETCD_IMAGE:-2.2.1}

shyamjvs replied:
Writing to a file, so changing it to an evaluable expression won't work.

shyamjvs commented:
@k8s-bot gci gke e2e test this

gmarek commented:
Looks good - let's merge it tomorrow morning, so we'll be able to monitor status of the SQ.

---

Reviewed 7 of 7 files at r7.
Review status: all files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/38227#-:-KYnZtoyWbmvmFhT4Z8l:bntt3gj)*
<!-- Sent from Reviewable.io -->
